upstream_utils/libdogleg.py

```diff
@@ -0,0 +1,39 @@
+#!/usr/bin/env python3
+
+import os
+import shutil
+
+from upstream_utils import Lib, walk_cwd_and_copy_if
+
+
+def copy_upstream_src(wpilib_root):
+    wpical = os.path.join(wpilib_root, "wpical")
+
+    # Delete old install
+    for d in [
+        "src/main/native/thirdparty/libdogleg/src",
+        "src/main/native/thirdparty/libdogleg/include",
+    ]:
+        shutil.rmtree(os.path.join(wpical, d), ignore_errors=True)
+
+    walk_cwd_and_copy_if(
+        lambda dp, f: f.endswith("dogleg.h"),
+        os.path.join(wpical, "src/main/native/thirdparty/libdogleg/include"),
+    )
+    walk_cwd_and_copy_if(
+        lambda dp, f: f.endswith("dogleg.c"),
+        os.path.join(wpical, "src/main/native/thirdparty/libdogleg/src"),
+    )
```

Comment:
```suggestion
    files = walk_cwd_and_copy_if(
        lambda dp, f: f.endswith("dogleg.h"),
        os.path.join(wpical, "src/main/native/thirdparty/libdogleg/include"),
    )
    for f in files:
        with open(f) as file:
            content = file.read()
        content = content.replace(
            "#include <cholmod.h>", "#include <suitesparse/cholmod.h>"
        )
        with open(f, "w") as file:
            file.write(content)

    files = walk_cwd_and_copy_if(
        lambda dp, f: f.endswith("dogleg.c"),
        os.path.join(wpical, "src/main/native/thirdparty/libdogleg/src"),
    )
    for f in files:
        with open(f) as file:
            content = file.read()
        content = content.replace("#warning", "// #warning")
        with open(f, "w") as file:
            file.write(content)
```
This should most of the MSVC incompatibilities as well fix the include path.